### ISSUE TITLE
Do not shade Byte Buddy

### ISSUE THREAD

**Issue by raphw** [assertj/assertj-core#2470]
Byte Buddy is currently shaded. As Byte Buddy retains compatibility, not shading the library would allow running recent JVMs without an update of assertj. Other libraries like Mockito or Hibernate do not shade BB and there are no known issues with this approach. In case of an error, stack traces would also be easier to process as break points remain intact.

**Comment by scordio** [comment]
Seems reasonable to me. What do you think @joel-costigliola?

**Comment by joel-costigliola** [comment]
If BB retains compatibility then I think it's fine, happy to remove shading as it will make life easier.

**Comment by joel-costigliola** [comment]
@raphw do you plan to have a major release at some point in the future that would break the backward compatibility?

**Comment by raphw** [comment]
No, I do not and given the wide spread of the library, I think I would change the namespace and group id in such a case.

**Comment by joel-costigliola** [comment]
Thanks @raphw! 

I think this xkcd is quite close to be true for byte buddy! 😆 (you might consider relocating to Nebraska!)
![image](https://user-images.githubusercontent.com/382613/151253752-7a59ad65-783f-4feb-9452-54813d953ead.png)
